File: jenkins-plugin/src/main/java/org/eazyportal/plugin/release/jenkins/action/PrepareRepositoryForReleaseActionFactory.java
package org.eazyportal.plugin.release.jenkins.action;

import com.google.inject.Inject;
import hudson.Extension;
import hudson.model.InvisibleAction;
import org.eazyportal.plugin.release.core.action.PrepareRepositoryForReleaseAction;
import org.eazyportal.plugin.release.jenkins.ReleaseStepConfig;

import java.io.Serializable;

@Extension
public class PrepareRepositoryForReleaseActionFactory extends InvisibleAction implements Serializable {

    @Inject
    private transient ReleaseStepConfig releaseStepConfig;

    public PrepareRepositoryForReleaseAction create() {
        return new PrepareRepositoryForReleaseAction(
            releaseStepConfig.getScmActions(),
            releaseStepConfig.getScmConfig()
        );
    }

}

File: gradle-plugin/src/main/kotlin/org/eazyportal/plugin/release/gradle/action/PrepareRepositoryForReleaseActionFactory.kt
package org.eazyportal.plugin.release.gradle.action

import org.eazyportal.plugin.release.core.action.PrepareRepositoryForReleaseAction
import org.eazyportal.plugin.release.gradle.model.EazyReleasePluginExtension

class PrepareRepositoryForReleaseActionFactory {

    fun create(extension: EazyReleasePluginExtension): PrepareRepositoryForReleaseAction =
        PrepareRepositoryForReleaseAction(
            extension.scmActions,
            extension.scmConfig
        )

}